packages/webview_flutter/ios/Classes/FlutterWebView.m

```diff
@@ -271,6 +275,30 @@ - (void)updateJsMode:(NSNumber*)mode {
   }
 }
 
+- (void)updateAutoMediaPlaybackPolicy:(NSNumber*)policy
+                      inConfiguration:(WKWebViewConfiguration*)configuration {
+  switch ([policy integerValue]) {
+    case 0:  // require_user_action_for_all_media_types
+      NSLog(@"requiring user action for all types");
+      if (@available(iOS 10.0, *)) {
+        configuration.mediaTypesRequiringUserActionForPlayback = WKAudiovisualMediaTypeAll;
+      } else {
+        configuration.mediaPlaybackRequiresUserAction = true;
+      }
+      break;
+    case 1:  // always_allow
+      NSLog(@"allowing auto playback");
+      if (@available(iOS 10.0, *)) {
+        configuration.mediaTypesRequiringUserActionForPlayback = WKAudiovisualMediaTypeNone;
+      } else {
+        configuration.mediaPlaybackRequiresUserAction = false;
+      }
+      break;
+    default:
+      NSLog(@"webview_flutter: unknown auto media plaback policy: %@", policy);
```

Comment:
nit: The Java side will silently eat any value instead of doing nothing and logging. I think either way is fine but it would be easier to understand if they were consistent.

Reply:
The iOS webview allows for a more fine grained control over this policy (e.g you can say that only audio or only video requires a user action) while the Android webview just lets you set a boolean.

The Android implementation is thus set to only allow media playback if the value is `always_allow`. The iOS implementation is built with this switch as if we decide to add more enum values we should handle each of them separately.

Reply:
Ah okay, makes sense. Thanks.